File: SpringTILCode/src/main/java/com/example/springtilcode/domain/post/controller/dto/response/PostResponse.java
package com.example.springtilcode.domain.post.controller.dto.response;

import com.example.springtilcode.domain.user.entity.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PostResponse {
    private String title;
    private String content;
    private User user;
}

File: kotlin/src/main/kotlin/com/example/demo/domain/post/controller/dto/response/PostResponse.kt
package com.example.demo.domain.post.controller.dto.response

import com.example.demo.domain.post.entity.Post
import com.example.demo.domain.user.entity.User

data class PostResponse(
    val title:String?,
    val content: String?,
    val user: User?
){
    constructor(post: Post): this(
        title = post.title,
        content = post.content,
        user = post.user
    )
}